frontend/src/features/admin-form/create/logic/components/LogicContent/EditLogicBlock/EditLogicBlock.tsx

```diff
@@ -0,0 +1,150 @@
+import { useCallback, useLayoutEffect, useMemo, useRef } from 'react'
+import { useFieldArray, useForm } from 'react-hook-form'
+import { Stack } from '@chakra-ui/react'
+import { merge } from 'lodash'
+
+import { useAdminLogicStore } from '../../../adminLogicStore'
+import { useAdminFormLogic } from '../../../hooks/useAdminFormLogic'
+import { useLogicMutations } from '../../../mutations'
+import { EditLogicInputs } from '../../../types'
+
+import {
+  AddConditionDivider,
+  EditConditionBlock,
+  EditConditionBlockDivider,
+  EditConditionWrapper,
+  SaveActionGroup,
+  ThenShowBlock,
+} from './EditCondition'
+
+export interface UseEditLogicBlockProps {
+  /** Sets default values of inputs if this is provided */
+  defaultValues?: Partial<EditLogicInputs>
+  onSubmit: (inputs: EditLogicInputs) => void
+}
+
+export const useEditLogicBlock = ({
+  defaultValues,
+  onSubmit,
+}: UseEditLogicBlockProps) => {
+  const setToInactive = useAdminLogicStore(
+    useCallback((state) => state.setToInactive, []),
+  )
+  const { logicableFields, mapIdToField, formFields } = useAdminFormLogic()
+  const { createLogicMutation } = useLogicMutations()
+
+  const formMethods = useForm<EditLogicInputs>({
+    defaultValues: merge({ conditions: [{}] }, defaultValues),
+  })
+  const {
+    fields: logicConditionBlocks,
+    append,
+    remove,
+  } = useFieldArray({
+    control: formMethods.control,
+    name: 'conditions',
+  })
+
+  const wrapperRef = useRef<HTMLDivElement | null>(null)
+
+  useLayoutEffect(() => {
+    if (wrapperRef.current) {
+      wrapperRef.current.scrollIntoView({
+        behavior: 'smooth',
+        // Block required so parent (with overflow:hidden) will not be scrolled
+        // and causing unscrollable white space.
+        // See https://stackoverflow.com/questions/48634459/scrollintoview-block-vs-inline/48635751#48635751
+        block: 'nearest',
+      })
+    }
+  }, [])
+
+  const handleAddCondition = useCallback(() => append({}), [append])
+
+  // Only allow logic removal if there is more than one logic block.
+  const handleRemoveCondition = useMemo(
+    () => (logicConditionBlocks.length > 1 ? remove : undefined),
```

Comment:
When we remove the condition block, we should clear the values as well, so when it gets re-added, it's back to empty.

Reply:
06c1347